Updated MotionControllerTest scene with debug display labels

cre8ivepark opened the pull request:
Updated MotionControllerTest scene with debug display text labels

![2017-09-14 18_21_02-mixed reality portal](https://user-images.githubusercontent.com/13754172/30462636-dc6264b8-997b-11e7-984a-c52277c9d14c.png)

Added motion controller 3D model related information text too.
![2017-09-14 23_55_00-unity 2017 2 0b11 personal 64bit - motioncontrollertest unity - holotoolkit-un 2](https://user-images.githubusercontent.com/13754172/30470513-e7653f9c-99a9-11e7-847e-bae14c98bf3f.png)

StephenHodgson commented:
looking good!

keveleigh commented on Assets/HoloToolkit/Utilities/Scripts/DebugPanelControllerInfo.cs:
```diff
@@ -104,6 +134,40 @@ private string GetControllerInfo()
                                           controllerState.TouchpadTouched, controllerState.TouchpadPosition);
+
+                // Text label display
+                if(controllerState.Handedness.ToString().Equals("Left"))
```
It'd probably be good to check controllerState.Handedness.Equals(InteractionSourceHandedness.Left) here instead of a string comparison (and the right controller below).

keveleigh commented on Assets/HoloToolkit/Utilities/Scripts/DebugPanelControllerInfo.cs:
```diff
@@ -28,6 +28,35 @@ private class ControllerState
 
+        // Text display label game objects
+        [SerializeField]
+        public GameObject leftInfoTextPointerPosition;
```
Can these be TextMesh instead of GameObject? It'd save having to do all the GetComponent<TextMesh>() calls below.

keveleigh commented on Assets/HoloToolkit/Utilities/Scripts/DebugPanelControllerInfo.cs:
```diff
@@ -28,6 +28,35 @@ private class ControllerState
         }
 
+        // Text display label game objects
+        [SerializeField]
```
Is SerializeField needed for public variables?

keveleigh commented on Assets/HoloToolkit/Utilities/Scripts/DebugPanelControllerInfo.cs:
```diff
@@ -94,6 +123,7 @@ private string GetControllerInfo()
             foreach (ControllerState controllerState in controllers.Values)
             {
+                // Debug message
                 toReturn += string.Format("Hand: {0}\nPointer: Position: {1} Rotation: {2}\n" +
```
This still shows up, causing duplicate info across the text panels (though, interestingly, I don't see it in any of your screenshots). Should we remove it in favor of yours?

DDReaper replied:
Daft question @cre8ivepark .  Why is this all using the older "TextMesh" components rather than the new UI system?  Just seems odd.

StephenHodgson replied:
@DDReaper really either can be used. It makes no difference.

StephenHodgson replied:
It's more of a performance issue.

cre8ivepark commented:
Thanks @keveleigh ! Updated the code and skybox based on your feedback. For the debug panel, I think it would be helpful to keep original one too, since it displays other useful messages including controller detect/lost or errors. I have added simple gradient cube skybox textures.
Updated screen:
![2017-09-15 14_29_10-mixed reality portal](https://user-images.githubusercontent.com/13754172/30504361-23862e68-9a23-11e7-8197-49e505732cf1.png)

keveleigh commented on Assets/HoloToolkit/Utilities/Scripts/DebugPanelControllerInfo.cs:
```diff
@@ -28,6 +28,34 @@ private class ControllerState
         }
 
+        // Text display label game objects
+        public TextMesh leftInfoTextPointerPosition;
```
I didn't catch this last time, but please capitalize the first letter of all the public variables here.

keveleigh commented on Assets/HoloToolkit/Input/Tests/Scenes/MotionControllerTest.unity:
```diff
@@ -607,6 +7105,347 @@ Prefab:
         type: 2}
+    - target: {fileID: 114710303647202208, guid: d29bc40b7f3df26479d6a0aac211c355,
+        type: 2}
+      propertyPath: RightControllerOverride
```
Please revert this override. My intention was to show one override and one controller model in the test scene.

cre8ivepark replied:
Oh, I see. I'll update. I'll add some details about this(game mode vs deploy) to the README too.

keveleigh commented:
@cre8ivepark Awesome, looks great! I just added two more comments, but I think that'll be the end of my review.

keveleigh commented:
Last thing: it looks like you may have applied some changes to the MixedRealityCameraParent prefab that should stay only in the scene. If you revert that file, this looks good to me!

StephenHodgson commented:
This looks great, but I won't be able to test until Monday when I get back into the office.

StephenHodgson commented:
Just out of curiosity, what is preventing us from loading the glTF models in the editor?

keveleigh commented:
@StephenHodgson There isn't a Unity API to access the models, so we have to call UWP APIs, which aren't supported in the editor.
